src/main/labview/labview.ts

```diff
@@ -222,6 +223,87 @@ export class LabView implements IDisposable {
     }
   }
 
+  get uiMode(): UIMode {
+    return this._uiMode;
+  }
+
+  async setUIMode(uiMode: UIMode) {
+    if (uiMode === UIMode.Custom) {
+      this._uiMode = uiMode;
+      return;
+    }
+
+    await this._setUIMode(uiMode);
+  }
+
+  async _setUIMode(uiMode: UIMode) {
+    this._uiMode = uiMode;
+
+    await this._view.webContents.executeJavaScript(`
+    executeJSResult_currentLayout = {};
+    {
+      const lab = window.jupyterapp || window.jupyterlab;
```

Comment:
I wonder if we should make the code below into a proper Jupyter extension and make it expose a single command on the window, like `window.setJupyterLabMode('zen' | 'multi' | 'single')`. This would ensure that as we update versions the type validation catches any changes, and allow re-using it elsewhere. This could be an extension or set of extensions in this repo, or even plugins which might be disabled by default in jupyterlab repo.

My main motivation would be to ultimately revert the change which exposed `jupyterapp` on window e.g. to avoid users writing logic which assumes that it is there (it will not be available outside of Desktop).

Reply:
that makes sense. I was also thinking about exposing desktop API endpoints (something like window.jlabDesktopApp.<methods>), so that extensions can use. I will create a separate issue for such refactoring.